pysmiles/write_smiles.py

```diff
@@ -187,3 +189,38 @@ def keyfunc(idx):
 
     smiles += ')' * branch_depth
     return smiles
+
+def write_smiles(molecule, default_element='*', start=None):
+    """
+    Creates a SMILES string describing `molecule` according to the OpenSMILES
+    standard. If `molecule` consists of multiple disconnected components their
+    corresponding SMILES string will be joined by zero-order bonds (".").
+
+    Parameters
+    ----------
+    molecule : nx.Graph
+        The molecule for which a SMILES string should be generated.
+    default_element : str
+        The element to write if the attribute is missing for a node.
+    start : Hashable
+        The atom at which the depth first traversal of the molecule should
+        start. A sensible one is chosen: preferably a terminal heteroatom.
+
+    Returns
+    -------
+    str
+        The SMILES string describing `molecule`.
+    """
+    smiles = []
+    for nodes in nx.connected_components(molecule):
+        if start and start in nodes:
+            start_ = start
+        else:
+            start_ = None
+        smiles.append(write_smiles_component(molecule.subgraph(nodes),
+                      default_element=default_element, start=start_))
+    if len(smiles) > 1:
+        LOGGER.info('Be aware the specified molecule is disconnected and '
+                    'consists of %d connected components.', len(smiles))
+    return '.'.join(smiles)
```

Comment:
I'd add this line to make the order of the smiles string deterministic and to ensure that the same networkx graph (based on isomorphism of each component) will always output the same smiles string, independent of the order of the components.
```suggestion
    smiles = sorted(smiles)       # make output deterministic and independent of order of fragments
    return '.'.join(smiles)
```

Reply:
I like the idea, but this won't be enough since the smiles will depend on the starting node, which in turn depends on the node keys in the graph.
I *could* try to sort the graph components based on minimal node keys, and revert to the usual, default ordering if the node keys are not comparable types.

Reply:
Ah yeah I see what you mean. The sorting of the components also doesn't really help here since just relabeling the graph with multiple components might change the smiles order. IF the smiles would be independent of the node ordering, sorting the smiles afterwards would ensure repeatable output, but sorting the components before, as now, would not. However, I think this is in total not a big issue since people can always set a random seed, so we can just leave it as is.